lock-api/src/main/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtils.java

```diff
@@ -0,0 +1,168 @@
+        return ConjureLockDescriptorListChecksum.of(checksum.type().getId(), Bytes.from(checksum.value()));
+    }
+
+    private static KeyListChecksum checksumFromConjure(ConjureLockDescriptorListChecksum conjureChecksum) {
+        return KeyListChecksum.of(
+                ChecksumType.valueOf(conjureChecksum.getTypeId()),
+                conjureChecksum.getValue().asNewByteArray());
+    }
+
+    @Unsafe
+    @Value.Immutable
+    public interface ConjureMetadataConversionResult {
+        List<LockDescriptor> lockList();
+
+        ConjureLockRequestMetadata conjureMetadata();
+
+        static ImmutableConjureMetadataConversionResult.Builder builder() {
+            return ImmutableConjureMetadataConversionResult.builder();
+        }
+    }
+
+    private enum ChangeMetadataToConjureVisitor implements ChangeMetadata.Visitor<ConjureChangeMetadata> {
+        INSTANCE;
+
+        @Override
+        public ConjureChangeMetadata visit(Unchanged unchanged) {
+            return ConjureChangeMetadata.unchanged(ConjureUnchangedChangeMetadata.of());
+        }
+
+        @Override
+        public ConjureChangeMetadata visit(Updated updated) {
+            return ConjureChangeMetadata.updated(
+                    ConjureUpdatedChangeMetadata.of(Bytes.from(updated.oldValue()), Bytes.from(updated.newValue())));
+        }
+
+        @Override
+        public ConjureChangeMetadata visit(Deleted deleted) {
+            return ConjureChangeMetadata.deleted(ConjureDeletedChangeMetadata.of(Bytes.from(deleted.oldValue())));
+        }
+
+        @Override
+        public ConjureChangeMetadata visit(Created created) {
+            return ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(created.newValue())));
+        }
+    }
+
+    private enum ChangeMetadataFromConjureVisitor implements ConjureChangeMetadata.Visitor<Optional<ChangeMetadata>> {
+        INSTANCE;
+
+        @Override
+        public Optional<ChangeMetadata> visitUnchanged(ConjureUnchangedChangeMetadata unchanged) {
+            return Optional.of(ChangeMetadata.unchanged());
+        }
+
+        @Override
+        public Optional<ChangeMetadata> visitUpdated(ConjureUpdatedChangeMetadata updated) {
+            return Optional.of(ChangeMetadata.updated(
+                    updated.getOldValue().asNewByteArray(),
+                    updated.getNewValue().asNewByteArray()));
+        }
+
+        @Override
+        public Optional<ChangeMetadata> visitDeleted(ConjureDeletedChangeMetadata deleted) {
+            return Optional.of(ChangeMetadata.deleted(deleted.getOldValue().asNewByteArray()));
+        }
+
+        @Override
+        public Optional<ChangeMetadata> visitCreated(ConjureCreatedChangeMetadata created) {
+            return Optional.of(ChangeMetadata.created(created.getNewValue().asNewByteArray()));
+        }
+
+        @Override
+        public Optional<ChangeMetadata> visitUnknown(String unknownType) {
+            return Optional.empty();
```

Comment:
It may be worth logging in this case, just so we know that the value is actively being dropped? Could potentially make it `TRACE`, or could do a rate limit log.